xmtp_mls/src/subscriptions/process_welcome.rs

```diff
@@ -159,13 +164,64 @@ where
         self.filter(process_result).await
     }
 
-    /// Applies conversation type filtering to processed welcome results.
+    /// Checks whether a group should be included in the stream based on filtering rules.
     ///
-    /// After a welcome message or group has been processed, this function determines
-    /// whether it should be streamed to the client based on:
+    /// Returns `Ok(true)` if the group should be included in the stream,
+    /// `Ok(false)` if it should be filtered out.
+    ///
+    /// # Arguments
+    /// * `group` - The group to check
+    /// * `check_virtual` - Whether to filter out virtual groups (only for new welcomes)
+    ///
+    /// # Filtering Rules
+    /// 1. Virtual groups are filtered out only if `check_virtual` is true
+    /// 2. Duplicate DMs are filtered out if `include_duplicate_dms` is false
+    /// 3. Conversation type must match if a filter is specified
+    /// 4. Consent state must match if a filter is specified
+    ///
+    /// # Errors
+    /// Returns an error if retrieving group metadata or consent state fails
+    async fn should_include_group(
```

Comment:
nit: `should_include_group_in_stream` or `should_include_in_stream`(to cover dms as well) to make it more clear?